src/hotspot/share/runtime/icache.cpp

```diff
@@ -24,24 +24,39 @@
 
 #include "code/codeBlob.hpp"
 #include "memory/resourceArea.hpp"
+#include "runtime/flags/flagSetting.hpp"
+#include "runtime/globals_extension.hpp"
 #include "runtime/icache.hpp"
+#include "runtime/java.hpp"
 #include "utilities/align.hpp"
 
 // The flush stub function address
 AbstractICache::flush_icache_stub_t AbstractICache::_flush_icache_stub = nullptr;
 
-void AbstractICache::initialize() {
+void AbstractICache::initialize(int phase) {
```

Comment:
Please add comment explaining different phases.

Reply:
Added. Also reshuffled a bit of code to make avoid unnecessary arch changes.